### PR TITLE
DNS request logging 

### PR DESCRIPTION
**Pull request by aschnapp**

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{
+    "version": "0.2.0",
+    "configurations": [
+        {
+            "type": "pwa-node",
+            "request": "launch",
+            "name": "Launch Program",
```
Comment (mfucci) [inline .vscode/launch.json:7]: "Launch daemon" ?
Or "Start backend" ?

File: src/frontend/components/DnsLogs.tsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+                <TableCell>Time</TableCell>
+                <TableCell>Type</TableCell>
+                <TableCell>Domain</TableCell>
+              </TableRow>
+            </TableHead>
+            <TableBody>
+              <Records<NetworkEventLog<DnsEvent>> dbName={NETWORK_EVENT_DATABASE}>
+                {(logs) => <Iterate array={logs}>{
+                  (log) => {
+                    if (log.device_id === this.props.id && log.service === 'DNS') return (
```
Comment (mfucci) [inline src/frontend/components/DnsLogs.tsx:33]: const { id }  = this.props; at the top of the function

File: src/frontend/components/DnsLogs.tsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+                      <TableRow key={log?.timestamp}>
+                        <TableCell>{moment(log?.timestamp).format('YYYY-MM-DD hh:mm:ss a')}</TableCell>
+                        <TableCell>{log?.event?.type}</TableCell>
+                        <TableCell>{log?.event?.name}</TableCell>
+                      </TableRow>
+                    )
+                    return null;
+                  }
+                }</Iterate>}
+              </Records>
```
Comment (mfucci) [inline src/frontend/components/DnsLogs.tsx:43]: } </Iterate>
} </Records>

File: src/frontend/components/DnsLogs.tsx [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+          <Table stickyHeader size="small" aria-label="Dns Log table">
+            <TableHead>
+              <TableRow>
+                <TableCell>Time</TableCell>
+                <TableCell>Type</TableCell>
+                <TableCell>Domain</TableCell>
+              </TableRow>
+            </TableHead>
+            <TableBody>
+              <Records<NetworkEventLog<DnsEvent>> dbName={NETWORK_EVENT_DATABASE}>
```
Comment (mfucci) [inline src/frontend/components/DnsLogs.tsx:30]:
```
<Records<NetworkEventLog<DnsEvent>> dbName={NETWORK_EVENT_DATABASE}> {logs =>
     <Iterate array={logs}>{log => {
```

File: src/frontend/components/DnsLogs.tsx (hunk 2 above)
Comment (mfucci) [inline src/frontend/components/DnsLogs.tsx:33]:
```
<If condition={log.device_id === this.props.id && log.service === 'DNS'}>
    (...)
</If>
```

File: src/backend/services/dns/DNSServer.ts [hunk 5]
```diff
@@ -7,33 +7,51 @@
...
     this.dnsMessenger.start()
       .then(server => console.log(`DNS server listening on ${server.address}:${server.port}...`))
       .catch(console.error);
   }
 
   private async handleRequest(request: Request) {
-    const { name, type } = request;
-
+    const { name, type, query: { _client: {family, address} } } = request;
+    const devices = await this.databaseManager.withDatabase<Device, Device[]>(DEVICES_DATABASE, async database => await database.getRecords())
```
[the remark this answers is not in the document]
Comment (aschnapp) [reply src/backend/services/dns/DNSServer.ts:46]: There is no databaseManager.getRecords. There is getRecord, but that gets a single record. Do you want me to add this function?

[the remark this answers is not in the document]
Comment (mfucci) [reply src/backend/services/dns/DNSServer.ts:47]: Yes, add it: 

```
    async getRecords<T>(name: string) {
        return await this.withDatabase<T, T[]>(name, database => database.getRecords());
    }
```

**Pull request merged**